lib/language_pack/ruby.rb

```diff
@@ -694,19 +694,32 @@ def build_bundler
       end
 
       if bundler.windows_gemfile_lock?
-        log("bundle", "has_windows_gemfile_lock")
-
-        File.unlink("Gemfile.lock")
-        ENV.delete("BUNDLE_DEPLOYMENT")
-
-        warn(<<~WARNING, inline: true)
-          Removing `Gemfile.lock` because it was generated on Windows.
-          Bundler will do a full resolve so native gems are handled properly.
-          This may result in unexpected gem versions being used in your app.
-          In rare occasions Bundler may not be able to resolve your dependencies at all.
-
-          https://devcenter.heroku.com/articles/bundler-windows-gemfile
-        WARNING
+        if bundler.supports_multiple_platforms?
+          puts "Windows `Gemfile.lock` detected, bundler #{@bundler.version} supports multiple platforms, no action taken."
+        else
+            File.unlink("Gemfile.lock")
```

Comment:
does there still need to be a log bundle here?

Reply:
That's a good question.

The mechanics behind this logging have been removed/disabled. We didn't clean them up because...it wasn't a priority and frankly I was worried about breaking something. Last time it was touched was https://github.com/heroku/heroku-buildpack-ruby/pull/1229 (found via git blame on GitHub's UI).

That `log` method seems to be defined here https://github.com/heroku/heroku-buildpack-ruby/blob/cf25ace0e09675e744603962e12451943ce979be/lib/language_pack/base.rb#L135 which calls https://github.com/heroku/heroku-buildpack-ruby/blob/cf25ace0e09675e744603962e12451943ce979be/lib/language_pack/base.rb#L217-L220 which shells out to `logger`. And that code was added by David Dollar in 2011, https://github.com/heroku/heroku-buildpack-ruby/commit/42a604366e609658971bd6bc56b32b06a964e057.

We're not currently using this functionality, so I think we should get rid of it. I will open an issue.

Our internal logging capabilities are in flux (constantly). I believe we're stabilizing on otel? soon when that happens, I think we should add logging to answer focused questions rather than expecting logging strings from 2011 to mean anything. We'll likely want to log windows builds, but we can add that functionality in when it comes time.

And FWIW we have other ways to get metrics for builds outside of these capabilities (so it's not as though we're operating with zero information). Buildpack logging helps to make getting those metrics easier and can allow us to collect targeted information as needed.

Reply:
https://github.com/heroku/heroku-buildpack-ruby/issues/1470